test/integration/slack_test.exs

```diff
@@ -61,4 +61,58 @@ defmodule Integration.SlackTest do
     message = send_message("@#{@bot}: operable:echo blah", private_channel)
     assert_response "blah", [after: message], private_channel
   end
+
+  test "redirecting from a private channel", %{user: user} do
+    user |> with_permission("operable:echo")
+    private_channel = "group_ci_bot_testing"
+    marker = send_message("Redirect test starts here")
```

Comment:
Yeah, I was trying to refactor as little as possible to keep the PR manageable. But I agree, the tests could use a bit of work.